data/crac-creation/crac-creator-cse/src/test/java/com/farao_community/farao/data/crac_creation/creator/cse/CseCracCreatorWithMneTest.java

```diff
@@ -0,0 +1,204 @@
+                            && (threshold.getUnit().equals(expectedThresholdUnit))
+            );
+        } else if (direction.equals("OPPOSITE")) {
+            return flowCnec.getThresholds().stream().anyMatch(threshold ->
+                    threshold.getSide().equals(side)
+                            && ((directionInvertedInNetwork && threshold.max().isPresent() && threshold.max().get().equals(expectedThreshold)) || (!directionInvertedInNetwork && threshold.min().isPresent() && threshold.min().get().equals(-expectedThreshold)))
+                            && (threshold.getUnit().equals(expectedThresholdUnit))
+            );
+        } else {
+            return flowCnec.getThresholds().stream().anyMatch(threshold ->
+                    threshold.getSide().equals(side)
+                            && threshold.max().isPresent() && threshold.max().get().equals(expectedThreshold)
+                            && threshold.min().isPresent() && threshold.min().get().equals(-expectedThreshold)
+                            && threshold.getUnit().equals(expectedThresholdUnit)
+            );
+        }
+    }
+
+    public void assertAllMneCorrectlyImportedInCrac() {
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_1", Instant.OUTAGE, 5000, 1.1, Unit.PERCENT_IMAX, 380);
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_1", Instant.AUTO, 5000, 6000, Unit.AMPERE, 380);
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_1", Instant.CURATIVE, 5000, 6500, Unit.AMPERE, 380);
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_2", Instant.OUTAGE, 5000, 1.1, Unit.PERCENT_IMAX, 380);
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_2", Instant.AUTO, 5000, 6000, Unit.AMPERE, 380);
+        assertMneWithContingencyInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", "outage_2", Instant.CURATIVE, 5000, 6500, Unit.AMPERE, 380);
+        assertMneBaseCaseInCrac("mne_test", "NNL3AA1", "NNL2AA1", "1", "DIRECT", Instant.PREVENTIVE, 5000, 1, Unit.PERCENT_IMAX, 380);
+    }
+
+    public void assertAllMneCorrectlyImportedInCriticalBranchesCreationContext() {
+        assertMneWithContingencyInCriticalBranchCreationContexts("mne_test", "outage_1", "NNL3AA1 ", "NNL2AA1 ", "1", true);
+        assertMneWithContingencyInCriticalBranchCreationContexts("mne_test", "outage_2", "NNL3AA1 ", "NNL2AA1 ", "1", true);
+        assertMneWithContingencyNotInCriticalBranchCreationContexts("mne_test - NNL3AA1  - NNL2AA1  - outage_3", ImportStatus.INCOMPLETE_DATA);
+        assertMneWithContingencyNotInCriticalBranchCreationContexts("fake_mne_1 - NNL1AA1  - NNL3AA1  - mneHasTooManyBranches", ImportStatus.NOT_YET_HANDLED_BY_FARAO);
```

Comment:
I think I prefere this to be an "inconsistent data" flag. since we won't ever handle this case (NOT_YET_HANDLED_BY_FARAO means it will be handled in the future)